src/formatDate.js

```diff
@@ -8,7 +8,40 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const dateInOldFormat = Array.from(fromFormat);
+  const newFormat = Array.from(toFormat);
+  const dateOnly = date.split(fromFormat[3]);
+
+  dateInOldFormat.pop();
+
+  for (let i = 0; i < dateInOldFormat.length; i++) {
+    dateInOldFormat[i] = dateOnly[i];
+  }
+
+  for (let i = 0; i < fromFormat.length - 1; i++) {
+    const dateEntityTemplate = fromFormat[i];
+
+    for (const el of toFormat) {
+      if (el.includes(dateEntityTemplate[0])) {
+        if (el.length === dateEntityTemplate.length) {
+          newFormat[newFormat.indexOf(dateEntityTemplate)] = dateInOldFormat[i];
+        } else {
+          if (el.length < dateEntityTemplate.length) {
+            newFormat[toFormat.indexOf(el)] = dateInOldFormat[i].slice(2);
+          } else {
+            newFormat[toFormat.indexOf(el)] =
+              dateInOldFormat[i] < 30
+                ? '20' + dateInOldFormat[i]
+                : '19' + dateInOldFormat[i];
```

Comment:
The logic for determining the century (`'20'` or `'19'`) is hardcoded and assumes that any year less than 30 is in the 2000s. This assumption may not hold true for all cases. Consider making this logic more flexible or configurable.